iothub/service/src/Amqp/AmqpSessionHandler.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Devices.Common.Exceptions;
+
+namespace Microsoft.Azure.Devices.Amqp
+{
+    /// <summary>
+    /// Handles a single AMQP session that holds the sender or receiver link that does the "work"
+    /// for the AMQP connection (receiving file upload notification, sending cloud to device messages, etc).
+    /// </summary>
+    internal class AmqpSessionHandler
+    {
+        private AmqpSession _session;
+        private AmqpSendingLinkHandler _sendingLinkHandler;
+        private AmqpReceivingLinkHandler _receivingLinkHandler;
+        private EventHandler _connectionLossHandler;
+
+        private string _linkAddress;
+
+        /// <summary>
+        /// Construct an AMQP session for handling sending cloud to device messaging, receiving file
+        /// upload notifications or receiving feedback messages.
+        /// </summary>
+        /// <param name="linkAddress">The link address to receive from/send to.</param>
+        /// <param name="connectionLossHandler">The handler to invoke if this session or its links are dropped.</param>
+        /// <param name="messageHandler">
+        /// The handler for received feedback messages or file upload notifications. If this session is used
+        /// for sending cloud to device messages, then this argument is ignored.
+        /// </param>
+        public AmqpSessionHandler(string linkAddress, EventHandler connectionLossHandler, Action<AmqpMessage> messageHandler = null)
+        {
+            _linkAddress = linkAddress;
+
+            _connectionLossHandler = connectionLossHandler;
+
+            if (_linkAddress.Equals(AmqpsConstants.FeedbackMessageAddress)
```

Comment:
Same question here about using `Equals()` instead of `==`.

Reply:
We can use == here since these are strings, sure.